src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -67,6 +73,124 @@ def create_resource_group(rg_name, location, tags=None):
+    ''' Gets a managed application.
+    :param str resource_group_name:the resource group name
+    :param str appliance_name:the managed application name
+    '''
+    racf = _resource_managedapps_client_factory()
+    if managedapp_id:
+        appliance = racf.appliances.get_by_id(managedapp_id)
+    else:
+        appliance = racf.appliances.get(resource_group_name, appliance_name)
+    return appliance
+
+
+def show_appliancedefinition(resource_group_name=None, appliance_definition_name=None,
+                             managedapp_definition_id=None):
+    ''' Gets a managed application definition.
+    :param str resource_group_name:the resource group name
+    :param str appliance_definition_name:the managed application definition name
+    '''
+    racf = _resource_managedapps_client_factory()
+    if managedapp_definition_id:
+        appliancedef = racf.appliance_definitions.get_by_id(managedapp_definition_id)
+    else:
+        appliancedef = racf.appliance_definitions.get(resource_group_name,
+                                                      appliance_definition_name)
+    return appliancedef
+
+
+def create_appliancedefinition(resource_group_name,  # pylint: disable=too-many-arguments
+                               appliance_definition_name, location,
+                               lock_level, package_file_uri, authorizations,
+                               description, display_name, tags=None):
+    ''' Create a new managed application definition.
+    :param str resource_group_name:the desired resource group name
+    :param str appliance_definition_name:the managed application definition name
+    :param str description:the managed application definition description
+    :param str display_name:the managed application definition display name
+    :param str package_file_uri:the managed application definition package file uri
+    :param str tags:tags in 'a=b c' format
+    '''
+    racf = _resource_managedapps_client_factory()
+    authorizations = authorizations or []
+    applianceAuthList = []
+
+    for name_value in authorizations:
+        # split at the first ':', both principalId and roldeDefinitionId should not have a ':'
+        principalId, roleDefinitionId = name_value.split(':', 1)
+        applianceAuth = ApplianceProviderAuthorization(principalId, roleDefinitionId)
+        applianceAuthList.append(applianceAuth)
+
+    applianceDef = ApplianceDefinition(lock_level, applianceAuthList, package_file_uri)
+    applianceDef.display_name = display_name
+    applianceDef.description = description
+    applianceDef.location = location
+    applianceDef.tags = tags
+
+    return racf.appliance_definitions.create_or_update(resource_group_name,
+                                                       appliance_definition_name, applianceDef)
+
+
+def list_appliances(resource_group_name=None):
+    racf = _resource_managedapps_client_factory()
+
+    if resource_group_name is not None:
```

Comment:
should be 
```
if resource_group_name:
    # use the resource group name
else:
    # list by subscription
```

When a string is empty, the expression `resource_group_name is not None` is `True`.